src/setuptools_betterproto/_command.py

```diff
@@ -59,53 +67,99 @@ class CompileProto(setuptools.Command):
 
         if not proto_files:
-            print(
-                f"No proto files found in {self.proto_path}/**/{self.proto_glob}/, "
-                "skipping compilation of proto files."
+            _logger.warning(
+                "No proto files found in %s with glob "
+                "%s, skipping compilation of proto files.",
+                self.config.proto_path,
+                self.config.proto_glob,
             )
             return
 
         protoc_cmd = [
             sys.executable,
             "-m",
             "grpc_tools.protoc",
-            *(f"-I{p}" for p in [self.proto_path, *include_paths]),
-            f"--python_betterproto_out={self.out_path}",
-            *map(str, proto_files),
+            *(f"-I{p}" for p in [self.config.proto_path, *self.config.include_paths]),
+            f"--python_betterproto_out={self.config.out_path}",
+            *proto_files,
         ]
 
-        print(f"Compiling proto files via: {' '.join(protoc_cmd)}")
+        _logger.info("compiling proto files via: %s", " ".join(protoc_cmd))
         subprocess.run(protoc_cmd, check=True)
 
 
-# This adds the build_betterproto command to the build sub-command.
-# The name of the command is mapped to the class name in the pyproject.toml file,
-# in the [project.entry-points.distutils.commands] section.
-# The None value is an optional function that can be used to determine if the
-# sub-command should be executed or not.
-_build_command.build.sub_commands.insert(0, ("build_betterproto", None))
+class AddProtoFiles(BaseProtoCommand):
+    """A command to add the proto files to the source distribution."""
+
+    def run(self) -> None:
+        """Copy the proto files to the source distribution."""
+        proto_files = self.config.expanded_proto_files
+        include_files = self.config.expanded_include_files
+
+        if include_files and not proto_files:
+            _logger.warning(
+                "Some proto files (%s) were found in the `include_paths` (%s), but "
+                "no proto files where found in the `proto_path` (%s). We are not adding "
+                "the found include files to the source distribution automatically!",
+                len(include_files),
+                ", ".join(self.config.include_paths),
+                self.config.proto_path,
+            )
+
+        if not proto_files:
+            return
+
+        dest_dir = self.distribution.get_fullname()
+
+        for file in (*proto_files, *include_files):
+            self.copy_with_path(file, os.path.join(dest_dir, file))
+
+        _logger.info("added %s proto files", len(proto_files) + len(include_files))
+
+    def copy_with_path(self, src: str, dest: str) -> None:
+        """Copy a file from src to dest, creating the destination directory."""
+        if not os.path.exists(os.path.dirname(dest)):
+            _logger.debug("creating directory %s", os.path.dirname(dest))
+            os.makedirs(os.path.dirname(dest))
```

Comment:
optionally, you might want to get the destination directory only once:
```suggestion
        dest_dir = os.path.dirname(dest)
        if not os.path.exists(dest_dir):
            _logger.debug("creating directory %s", dest_dir)
            os.makedirs(dest_dir)
```

Reply:
Yeah, good idea. I also updated the function name and improved the docs a bit.